src/nvim/syntax.c

```diff
@@ -3286,17 +3288,28 @@ static void syn_cmd_off(exarg_T *eap, int syncing)
 }
 
 static void syn_cmd_onoff(exarg_T *eap, char *name)
+  FUNC_ATTR_NONNULL_ALL
 {
-  char buf[100];
-
+  did_syntax_onoff = true;
   eap->nextcmd = check_nextcmd(eap->arg);
   if (!eap->skip) {
-    strcpy(buf, "so ");
+    char buf[100];
+    strncpy(buf, "so ", 3);
```

Comment:
Yep. I'll sleep on this. Passing a flag to the functions so they track they are being called because of this is another option, although not the cleanest.